core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -255,46 +230,49 @@ public void remove() {
   }
 
   /**
-   * Whether there is an object with the specified field value in the set.
+   * Whether there is an object with the specified index field value in the set.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value
    * @return true if there is one such object, otherwise false
    */
-  public boolean contains(FieldIndex<T> index, Object value) {
-    ConcurrentHashSet<T> set = getByFieldInternal(index, value);
-    return set != null && !set.isEmpty();
+  public boolean contains(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    if (index == null) {
+      throw new IllegalStateException("use an index that isn't defined in the IndexedSet");
```

Comment:
this sounds like you're asking the user to use an index that isn't defined in the IndexedSet. change this to "the given index isn't defined for this IndexedSet" here and in the other two instances